plugins/outputs/influxdb_v2/http.go

```diff
@@ -166,7 +183,33 @@ func (c *httpClient) Write(ctx context.Context, metrics []telegraf.Metric) error
 		return nil
 	}
 
-	desc := resp.Header.Get("X-Influx-Error")
+	writeResp := &genericRespError{}
+	json.NewDecoder(resp.Body).Decode(writeResp)
+	var desc string
+
+	switch resp.StatusCode {
+	case http.StatusBadRequest: // 400
+		// LineProtocolError
+		desc = fmt.Sprintf("%s - %s;%d;%s", writeResp, writeResp.Op, writeResp.Line, writeResp.Err)
+	case http.StatusUnauthorized, http.StatusForbidden: // 401, 403
+		// Error
+		desc = fmt.Sprintf("%s - %s;%s", writeResp, writeResp.Op, writeResp.Err)
+	case http.StatusRequestEntityTooLarge: // 413
+		// LineProtocolLengthError
+		desc = fmt.Sprintf("%s - %s;%d", writeResp, writeResp.Op, writeResp.MaxLength)
+	case http.StatusTooManyRequests, http.StatusServiceUnavailable: // 429, 503
+		retryAfter := resp.Header.Get("Retry-After")
+		retry, err := strconv.Atoi(retryAfter)
+		if err != nil {
+			return fmt.Errorf("Bad value for 'Retry-After': %s", err.Error())
+		}
+		time.Sleep(time.Second * time.Duration(retry))
+		c.Write(ctx, metrics)
+	}
+
+	if xErr := resp.Header.Get("X-Influx-Error"); xErr != "" {
+		desc = fmt.Sprintf("%s - %s", desc, xErr)
```

Comment:
because you are parsing the JSON error body, I don't think you need X-Influx-Error

Reply:
I meant to note to remove this once the spec was fully implemented